src/core/Akka/Actor/ActorBase.cs

```diff
@@ -155,6 +166,31 @@ protected static IActorContext Context
         /// <returns>TBD</returns>
         protected internal virtual bool AroundReceive(Receive receive, object message)
         {
+            if (message is Scheduler.TimerScheduler.ITimerMsg tm)
```

Comment:
This is hot path of message execution. I'd check if this somehow affects actor benchmarks.

Reply:
not sure if there is a better place for this to work in general.
Other option is do this only in specific actors where you want to use `Timers`, but this would be easy to forget

Reply:
What about this approach?
To use timers, you'll have to enable them in actor constructor via this method:
https://github.com/akkadotnet/akka.net/blob/16932f4561d46c75d68908b8b9bc97880daacb8f/src/core/Akka/Actor/ActorBase.cs#L162

and then we can avoid type check in actors not using timers, and only do a null check first:
https://github.com/akkadotnet/akka.net/blob/16932f4561d46c75d68908b8b9bc97880daacb8f/src/core/Akka/Actor/ActorBase.cs#L175-L177

Reply:
Why not use an interface similar to Scala's original trait?

```
if(this is IWithTimers wt){
     WithTimers();
}
```

Having users manually have to call that method is going to be a pain point - decorating the actor with an interface like how they do with stash is a good compromise. Signals the intent that they want to use this feature via the type system rather than an internal method call.

Reply:
updated, now actors which want to use timers need to implement `IWithTimers` interface

Reply:
Thanks, looks great